cache/disk/lru.go

```diff
@@ -120,6 +124,44 @@ func (c *sizedLRU) MaxSize() int64 {
 	return c.maxSize
 }
 
+func (c *sizedLRU) Reserve(size int64) bool {
+	if size < 0 || size > c.maxSize || c.reservedSize+size > c.maxSize {
```

Comment:
`if size < 0 || c.reservedSize+size > c.MaxSize`

When would size be < 0?

Reply:
This is a safety check for HTTP API request where Content-Length is unset/-1.

Reply:
Also shouldn't you be checking c.currentSize instead of c.reservedSize?

Reply:
Checking with `reservedSize` instead of `currentSize` is definitely a bug.

But just checking `c.currentSize+size > c.maxSize` for going over the upper bound is problematic if the result overflows int64. Maybe unlikely in practice, but `size` comes directly from the client. If the client sends an unusually value, then `size > c.maxSize` is likely to catch this, but I think I can come up with a more correct check.

Reply:
Hmm, wait... We shouldn't check `c.currentSize+size > c.maxSize` at the start, since we can evict items to make space.

This code was checking `c.reservedSize+size > c.maxSize` because if that fails, we can't evict items to reserve enough space. I will change that part back for now, while still considering how to handle overflow.

Reply:
Refactored this a bit to improve the overflow checking, and added a comment about why we check size+reservedSize.